### DISCUSSION TITLE
RBAC

### DISCUSSION THREAD

**Discussion by mamalli11** [Hossein-Falah/Store-Api#29] [Q&A] [answered]
سلام 🖐🏻
**اقا دمت گرم چه کد خفنی نوشتی 👌**
میشه لطف کنی منطق پیاده سازی RBAC را برام توضیح بدی 🙏🏻

**Comment by Hossein-Falah** [comment, marked as answer]
سلام محمد جواد جان❤️
مرسی ممنونم از نگاهت🙏
هدف من کلا از پیاده سازی این RBAC اینه که هر کاربر با دسترسی ها USER ADMIN AUTHOR ,
یه سری دسترسی ها داشته باشن و دسترسی بعضی ها محدود باشه که نتونن به بخش ها حساس دسترسی داشته باشن
مثلا به عنوان مثال سطح دسترسی AUTHOR فقط بتونه [CREATE_BLOG, UPDATE_BLOG, READ_BLOG] بتونه انجام بده چون منطقی نیست که سطح AUTHOR مثلا بتونه به بخش های حساس کاربر ها دسترسی داشته باشه یا بخش تیکت های پشتبانی که فقط ADMIN USER میتون به اون دسترسی داشته باش AUTHOR هم بتونه دسترسی داشته باشه این اصلا منطقی نیست
پس کلا من RBAC برای این پیاده سازی کردم هر کدوم از سطح ها بتونن کار های خاصی رو انجام بدن و AUTHOR USER محدود باشن هر کاری نتونن انجام بدن
امید وارم توضیحاتم برات مفید بوده باشه و به علم شما اضافه کرده باشه❤️😘

**Reply by mamalli11** [reply]
خیلی هم عالی سپاسگذارم از توضیحاتت ❤️🙏🏻

**Reply by Hossein-Falah** [reply]
خواهش میکنم❤️
